ctapipe_io_lst/tests/test_multifile.py

```diff
@@ -0,0 +1,54 @@
+import pytest
+from pathlib import Path
+import os
+
+test_data = Path(os.getenv('LSTCHAIN_TEST_DATA', 'test_data')).absolute()
+test_r0_dir = test_data / 'real/R0/20200218'
+
+
+def test_multifile():
+    from ctapipe_io_lst.multifiles import MultiFiles
+
+    paths = test_r0_dir.glob('LST-1.*.Run02005.0000_first50.fits.fz')
+
+    with MultiFiles(paths) as multi_files:
+
+        assert len(multi_files) == 200
+        assert multi_files.num_inputs() == 4
+
+        event_count = 0
+        for event in multi_files:
+            event_count += 1
+            assert event.event_id == event_count
+
+        assert event_count == 200
```

Comment:
I think this last check is redundant, since you already check that len(multi_files)=200, and you have a simple for loop above increasing a local variable this 200 times, I think no matter what MultiFiles does, you will always have event_count set to 200 at the end of the loop

Reply:
No, it's not redundant. Since `MultiFile` implements the `iterator` interface, `__len__` can do something different than iterating over it. Checking both tests the implementation of `__len__` and of `__iter__` and `__next__`

Reply:
ok, thx for explanation, resolving comment